package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,307 @@
+coordinates for a trajectory. This non-linear dimension reduction method
+assumes that the trajectory is long enough to represent a probability
+distribution of as protein close to the equilibrium. Furthermore, the diffusion
+map assumes that the diffusion coefficients are constant. The eigenvectors with
+the largest eigenvalues are the slowest collective coordinates. The complexity of
+the diffusion map is O(N^3), where N is the number of frames in the trajectory.
+Instead of a single trajectory a sample of protein structures can be used.
+The sample should be equiblibrated, at least locally.  Different weights can be
+used. The order of the sampled structures in the trajectory is irrelevant.
+
+The :ref:`Diffusion-Map-tutorial` shows how to use diffusion map for dimension
+reduction.
+
+More details about diffusion maps are in [Lafon1]_  and [Clementi1]_.
+
+.. _Diffusion-Map-tutorial:
+
+Diffusion Map tutorial
+--------------------
+
+The example uses files provided as part of the MDAnalysis test suite
+(in the variables :data:`~MDAnalysis.tests.datafiles.PSF` and
+:data:`~MDAnalysis.tests.datafiles.DCD`). Notice that this trajectory
+does not represent a sample from the equilibrium. This violates a basic
+assumption of the diffusion map and the results shouldn't be interpreted for this reason.
+This tutorial shows how to use the diffusionmap function.
+First load all modules and test data ::
+
+   >>> import MDAnalysis
+   >>> import numpy as np
+   >>> import MDAnalysis.analysis.diffusionmap as diffusionmap
+   >>> from MDAnalysis.tests.datafiles import PSF,DCD
+
+Given an universe or atom group, we can calculate the diffusion map from
+that trajectory using :class:`DiffusionMap`:: and get the corresponding eigenvalues
+and eigenvectors
+
+   >>> u = MDAnalysis.Universe(PSF,DCD)
+   >>> dmap = diffusionmap.DiffusionMap(u)
+   >>> dmap.run()
+   >>> eigenvalues = dmap.eigenvalues
+   >>> eigenvectors = dmap.eigenvectors
+
+Classes
+-------
+
+.. autoclass:: DiffusionMap
+   :members:
+
+References
+---------
+
+If you use this Dimension Reduction method in a publication, please
+reference:
+..[Lafon1] Coifman, Ronald R., Lafon, Stephane (2006) Diffusion maps.
+Appl. Comput. Harmon. Anal. 21, 5–30.
+..[Clementi1] Rohrdanz, M. A, Zheng, W, Maggioni, M, & Clementi, C. (2013)
+Determination of reaction coordinates via locally scaled
+diffusion map. Journal of Chemical Physics.
+
+.. If you choose the default metric, this module uses the fast QCP algorithm
+[Theobald2005]_ to calculate the root mean square distance (RMSD) between
+two coordinate sets (as implemented in :func:`MDAnalysis.lib.qcprot.CalcRMSDRotationalMatrix`).
+When using this module in published work please cite [Theobald2005]_.
+
+
+"""
+from six.moves import range
+import logging
+
+import numpy as np
+
+from MDAnalysis.analysis import rms
+
+from .base import AnalysisBase
+
+logger = logging.getLogger("MDAnalysis.analysis.diffusionmap")
+
+
+class DistMatrix(AnalysisBase):
```

Comment:
Okay, that clears things up, when initializing the diffusion map, I have to initialize a DistMatrix object before passing it as an argument to DiffusionMap() if this is the case, correct?